nucypher/network/protocols.py

```diff
@@ -96,4 +116,128 @@ def __radd__(self, other):
         return bytes(other) + bytes(self)
 
     def __repr__(self):
-        return self.uri
+	    return self.uri
+
+
+class HendrixResourceWithStatics(HendrixResource):
+
+    def getChild(self, name, request):
+
+        path = name.decode('utf-8')
+        if path in self.children:
+            return self.children[path]
+
+        return super().getChild(name, request)
+
+
+class HendrixTCPServiceWithTLSAndStatics(HendrixTCPServiceWithTLS):
+
+    def __init__(self, port, private_key, cert,
+        context_factory=None,
+        context_factory_kwargs=None,
+        application=None,
+        threadpool=None,
+    ):
+        hxresource = HendrixResourceWithStatics(
+
+            reactor, threadpool, application)
+
+        child = File(STATICS_DIR)
+        child.namespace = 'statics'
+        hxresource.putNamedChild(child)
+        site = server.Site(hxresource)
+
+        context_factory = context_factory or ssl.DefaultOpenSSLContextFactory
+        context_factory_kwargs = context_factory_kwargs or {}
+
+        self.tls_context = context_factory(
+            private_key,
+            cert,
+            **context_factory_kwargs
+        )
+        internet.SSLServer.__init__(
+            self,
+            port,  # integer port
+            site,  # our site object, see the web howto
+            contextFactory=self.tls_context
+        )
+
+
+class HendrixServiceWithStatics(HendrixService):
+
+    def __init__(
+            self,
+            application,
+            threadpool=None,
+            resources=None,
+            services=None,
+            loud=False):
+        service.MultiService.__init__(self)
+
+        # Create, start and add a thread pool service, which is made available
+        # to our WSGIResource within HendrixResource
+        if not threadpool:
+            self.threadpool = ThreadPool(name="HendrixService")
+        else:
+            self.threadpool = threadpool
+
+        reactor.addSystemEventTrigger('after', 'shutdown', self.threadpool.stop)
+        ThreadPoolService(self.threadpool).setServiceParent(self)
+
+        # create the base resource and add any additional static resources
+        resource = HendrixResourceWithStatics(reactor, self.threadpool, application, loud=loud)
+        if resources:
+            resources = sorted(resources, key=lambda r: r.namespace)
+            for res in resources:
+                if hasattr(res, 'get_resources'):
+                    for sub_res in res.get_resources():
+                        resource.putNamedChild(sub_res)
```

Comment:
woah nested!

Reply:
recurse this!

Reply:
Yes - please extract to another function to help break up this nesting!